src/Components/ArtworkAlert/Hooks/useArtworkAlertContext.tsx

```diff
@@ -0,0 +1,208 @@
+      }
+    }
+  | { type: "SET_SEARCH_CRITERIA_ID"; payload: string }
+
+const reducer = (onReset: () => State) => (state: State, action: Action) => {
+  switch (action.type) {
+    case "RESET":
+      return onReset()
+
+    case "SET_SETTINGS":
+      return {
+        ...state,
+        settings: action.payload,
+      }
+
+    case "SET_CRITERIA":
+      return {
+        ...state,
+        criteria: action.payload,
+      }
+
+    case "SET_CRITERIA_ATTRIBUTE":
+      return {
+        ...state,
+        criteria: {
+          ...state.criteria,
+          [action.payload.key]: action.payload.value,
+        },
+      }
+
+    case "REMOVE_CRITERIA_ATTRIBUTE_VALUE":
+      let criteriaValue = state.criteria[action.payload.key]
+
+      if (Array.isArray(criteriaValue)) {
+        criteriaValue = criteriaValue.filter(
+          currentValue => currentValue !== action.payload.value
+        )
+      } else {
+        criteriaValue = null
+      }
+
+      return {
+        ...state,
+        criteria: {
+          ...state.criteria,
+          [action.payload.key]: criteriaValue,
+        },
+      }
+
+    case "SET_SEARCH_CRITERIA_ID":
+      return {
+        ...state,
+        searchCriteriaID: action.payload,
+      }
+
+    default:
+      return state
+  }
+}
+
+const ArtworkAlertContext = createContext<{
+  current: "ALERT_DETAILS" | "ALERT_FILTERS" | "ALERT_CONFIRMATION"
+  dispatch: React.Dispatch<Action>
+  goToFilters(): void
+  goToDetails(): void
+  onComplete(): void
+  state: State
+}>({
+  current: "ALERT_DETAILS",
+  dispatch: () => {},
+  goToDetails: () => {},
+  goToFilters: () => {},
+  onComplete: () => {},
+  state: DEFAULT_STATE,
+})
+
+interface ArtworkAlertProviderProps {
+  initialCriteria?: SearchCriteriaAttributes
+  currentArtworkID?: string
+}
+
+export const ArtworkAlertProvider: FC<ArtworkAlertProviderProps> = ({
+  children,
+  initialCriteria,
+  currentArtworkID,
+}) => {
+  const initialState = {
+    settings: DEFAULT_STATE.settings,
+    criteria:
+      getAllowedSearchCriteria(initialCriteria as SearchCriteriaAttributes) ??
+      DEFAULT_STATE.criteria,
+    currentArtworkID,
+  }
+  const basis = useRef<State>(initialState)
+  const [current, setCurrent] = useState<
+    "ALERT_DETAILS" | "ALERT_FILTERS" | "ALERT_CONFIRMATION"
+  >("ALERT_DETAILS")
+  const { createSavedSearch } = useCreateSavedSearch()
+
+  const handleComplete = async () => {
+    try {
+      const reponse = await createSavedSearch({
+        attributes: state.criteria,
+        userAlertSettings: state.settings,
+      })
+      if (reponse.createSavedSearch?.savedSearchOrErrors.internalID) {
+        dispatch({
+          type: "SET_SEARCH_CRITERIA_ID",
+          payload: reponse.createSavedSearch?.savedSearchOrErrors.internalID,
+        })
+      }
+      setCurrent("ALERT_CONFIRMATION")
+    } catch (error) {
+      console.error("ArtworkAlert/useArtworkAlertContext", error)
+    }
+  }
```

Comment:
**walkthrough**: Very handy to be able to respond in this way when the alert details are ready to be submitted to the backend. We can persist the data, perform any error handling, and transition when we're ready — without needing the clients to get into the weeds in how that's done.